pilot/pkg/model/xds_cache.go

```diff
@@ -245,6 +251,11 @@ func (l *lruCache) Add(entry XdsCacheEntry, pushReq *PushRequest, value *discove
 			return
 		}
 		if l.enableAssertions {
+			for _, opt := range opts {
+				if opt.IgnoreOldCache {
+					continue
+				}
+			}
```

Comment:
@howardjohn to ignore cache assert for eds PA case

Reply:
good idea. I don't love it since we lose the check but I don't see a better option

Reply:
Me too.
Actually assert is only for tests to make sure the cache works well, so we cannot remove it either.